hera_pspec/grouping.py

```diff
@@ -288,74 +308,88 @@ def average_spectra(uvp_in, blpair_groups=None, time_avg=False,
+                    # including the error_filed in stats_array and cov_array. 
+                    # For each power spectrum P_i with error sigma_i, 
+                    # P_avg = \sum{ P_i / (sigma_i)^2 } / \sum{ 1 / (sigma_i)^2 } 
+                    # while for terms like variance or covariance,
+                    # epsilon_avg = \sum{ (epsilon_i / (sigma_i)^4 } / ( \sum{ 1 / (sigma_i)^2 } )^2
+                    # For reference: M. Tegmark 1997, The Astrophysical Journal Letters, 480, L87, Table 1, #3
+                    # or J. Dillon 2014, Physical Review D, 89, 023002 , Equation 34. 
+                        w = 1./(uvp.get_stats(error_weights, (spw, blp, p)))**2
+                        # shape of w: (Ntimes, Ndlys)
+                    else:
+                    # Otherwise all arrays are averaged in a way weighted by the integration time,
+                    # including the error_filed in stats_array and cov_array. 
+                        w = (ints * np.sqrt(nsmp))
+                        # shape of w: (Ntimes, 1)
+                    
                     # Take time average if desired
                     if time_avg:
-                      data = (np.sum(data * w, axis=0) \
-                           / np.sum(w, axis=0).clip(1e-10, np.inf))[None]
-                      wgts = (np.sum(wgts * w[:, None], axis=0) \
-                           / np.sum(w, axis=0).clip(1e-10, np.inf)[:, None])[None]
-                      ints = (np.sum(ints * w, axis=0) \
-                           / np.sum(w, axis=0).clip(1e-10, np.inf))[None]
-                      nsmp = np.sum(nsmp, axis=0)[None]
-                      w = np.sum(w, axis=0)[None]
-                      if store_cov:
-                          cov = (np.sum(cov * w[:, None], axis=0) \
-                          / np.sum(w,axis=0).clip(1e-10, np.inf))[None]
-
-                      for stat in stat_l:
-                        errws[stat] = np.sum(errws[stat], axis=0)[None]
+                        data = (np.sum(data * w, axis=0) \
+                            / np.sum(w, axis=0))[None]
+                        wgts = (np.sum(wgts * w[:, :, None], axis=0) \
+                            / np.sum(w, axis=0)[:, None])[None]
+                        ints = (np.sum(ints * w, axis=0) \
+                            / np.sum(w, axis=0))[None]
+                        nsmp = np.sum(nsmp, axis=0)[None]
+                        if store_cov:
+                            cov = (np.sum(cov * w[:, :, None]**2, axis=0) \
+                                / (np.sum(w,axis=0))[:, None]**2)[None]
+                        for stat in stat_l:
+                            errws[stat] = (np.sum(errws[stat]*w**2, axis=0) \
+                            / (np.sum(w, axis=0))**2)[None]
+                        w = np.sum(w, axis=0)[None]
                     # Add multiple copies of data for each baseline according
                     # to the weighting/multiplicity
```

Comment:
should add this note for new developers
`: multiple copies are only added when bootstrap resampling`